sympy/matrices/expressions/diagonal.py

```diff
@@ -1,14 +1,19 @@
 from __future__ import print_function, division
 
 from sympy.matrices.expressions import MatrixExpr
-from sympy.core import S
-
+from sympy.core import S,Eq
+from sympy.functions.special.tensor_functions import KroneckerDelta
 class DiagonalMatrix(MatrixExpr):
     arg = property(lambda self: self.args[0])
     shape = property(lambda self: (self.arg.shape[0], self.arg.shape[0]))
 
     def _entry(self, i, j):
-        return S.Zero if i != j else self.arg[i, 0]
+        eq=Eq(i,j)
```

Comment:
Yes, I believe that is wrong since a diagonal matrix does not mean that all diagonal elements are the same, only that all off-diagonal elements are 0.

But the comment meant to add a space after the comma in line 11 and a similar line below. Also, there should be spaces around the `=` of line 11.